backend/service/service_proposal.go

```diff
@@ -48,25 +63,240 @@ func (service *ProposalService) QueryProposalsByHeight(height int64) []model.Pro
 	return resp
 }
 
+func (service *ProposalService) QueryDepositAndVotingProposalList() []model.ProposalNewStyle {
+
+	var data []document.Proposal
+	sort := desc(document.Proposal_Field_SubmitTime)
+	selector := bson.M{
+		document.Proposal_Field_ProposalId:        1,
+		document.Proposal_Field_Title:             1,
+		document.Proposal_Field_Type:              1,
+		document.Proposal_Field_Status:            1,
+		document.Proposal_Field_Votes:             1,
+		document.Proposal_Field_VotingStartHeight: 1,
+		document.Proposal_Field_TotalDeposit:      1,
+	}
+
+	if err := queryAll(document.CollectionNmProposal, selector, bson.M{"status": bson.M{"$in": []string{document.ProposalStatusDeposit, document.ProposalStatusVoting}}}, sort, 0, &data); err != nil {
+		logger.Error("query proposal collection", logger.String("err", err.Error()))
+		return nil
+	}
+
+	heightArr := make([]int64, 0, len(data))
+	depositProposalIdArr := []uint64{}
+	voterAddrArr := make([]string, 0, len(data))
+
+	for _, v := range data {
+
+		if v.Status == document.ProposalStatusDeposit {
+			depositProposalIdArr = append(depositProposalIdArr, v.ProposalId)
+		}
+
+		if v.Status == document.ProposalStatusVoting {
+			heightArr = append(heightArr, v.VotingStartHeight)
+			for _, addr := range v.Votes {
+				tag := false
+				for _, vAddr := range voterAddrArr {
```

Comment:
delete remove duplicated code. voter addr is distinct in one proposal

Reply:
[how to delete duplicates in a slice](https://stackoverflow.com/questions/26172673/how-to-delete-duplicates-of-an-item-from-a-slice)